cli/azd/internal/agent/conversational_agent.go

```diff
@@ -178,3 +206,95 @@ func (aai *ConversationalAzdAiAgent) renderThoughts(ctx context.Context) (func()
 
 	return cleanup, canvas.Run()
 }
+
+func startWatcher(ctx context.Context, fileChanges *FileChanges, mu *sync.Mutex) (*fsnotify.Watcher, chan bool, error) {
```

Comment:
We should move the file watching semantics outside of the agent code then invoke it from here so this can be tested/used in isolation or in other areas.

We should also consider splitting out the rendering of the file changes from the `SendMessage(...)` call